python_orocos_kdl_vendor/patches/0001-fix-windows-debug.patch

```diff
@@ -0,0 +1,20 @@
+--- a/python_orocos_kdl/CMakeLists.txt	2023-12-08 15:19:53.469243534 +0000
++++ b/python_orocos_kdl/CMakeLists.txt	2023-12-08 15:20:18.110104659 +0000
+@@ -26,6 +26,17 @@ endif()
+ set(PYBIND11_PYTHON_VERSION ${PYTHON_VERSION} CACHE STRING "Python version used by PyBind11")
+ 
+ find_package(Python ${PYTHON_VERSION} COMPONENTS Interpreter Development REQUIRED)
++if(WIN32 AND CMAKE_BUILD_TYPE STREQUAL "Debug")
++  get_filename_component(_python_executable_dir "${PYTHON_EXECUTABLE}" DIRECTORY)
```

Comment:
I think `PYTHON_EXECUTABLE` is the name from the deprecated [PythonInterp](https://cmake.org/cmake/help/latest/module/FindPythonInterp.html) module. Since this is using `find_package(Python ...)` on line 6 I think it [should use `Python_EXECUTABLE`](https://cmake.org/cmake/help/latest/module/FindPython.html#result-variables) instead.

Reply:
I wonder where it's getting `PYTHON_EXECUTABLE` from? I'm surprised by the CI results.

Reply:
I'm not 100% sure, but I suspect it is from `python_cmake_module`, which is `find_package`d in the `python_orocos_kdl_vendor` CMakeLists.txt.  Let me try switching this to `Python_EXECUTABLE` instead.

Reply:
And yep, that worked and is probably more correct.  @sloretz When you get a chance, I could use another review here, thanks!